src/ast/seq_decl_plugin.cpp

```diff
@@ -1463,3 +1463,191 @@ bool seq_util::re::is_epsilon(expr* r) const {
 app* seq_util::re::mk_epsilon(sort* seq_sort) {
     return mk_to_re(u.str.mk_empty(seq_sort));
 }
+
+/*
+  Provides a standard pretty printed view of the regex r when possible.
+*/
+std::string seq_util::re::pp(expr* r) {
+    SASSERT(u.is_re(r));
+    std::ostringstream buffer;
+    pp_compact_to_buffer(buffer, r);
+    return buffer.str();
+}
+
+void seq_util::re::pp_compact_to_buffer(std::ostringstream& buffer, expr* r) {
```

Comment:
std::ostream& is more general than std::ostringstream&

Reply:
will change